src/components/RewardList.js

```diff
@@ -33,6 +33,10 @@ const RewardList = ({ rewards }) => {
       return rewardDate >= start && rewardDate <= end;
     });
 
+  // this approach is wrong, sortedFilteredRewards should be some state, which will be updated whenever
+  // searchTerm and sortField changes which will be operated on the whole reward list (centrally managed)
+  // currently, these are being calculated everytime the component refreshes.
+  // same goes for start and end date
```

Comment:
changed my approach in the following way:-

filteredSortedRewards is initialized with the full rewards list.

Central Management with useEffect to update filteredSortedRewards whenever searchTerm, sortField, startDate, endDate, or rewards change. This ensures that the calculations are done only when necessary.